### PR TITLE
Uueoa-Esa, Part 3: Ruin Assets

### PR DESCRIPTION
**Pull request by RustingWithYou**
Adds several new items and structures for use in Unathi ruins. These include:
Several ancient Unathi clothing items, including two sets of armor.
Pillars and statues for ruin decoration.
Three ancient Unathi melee weapons.
Several flags/banners/tapestries.
Sarcophagi, complete with booby-traps.
An undetonated nuclear bomb, for, uh... don't worry about it.

Bomb sprite by @nauticall, other ruin sprites by Gecko

### CODE REVIEW

**Comment by RustingWithYou** [comment]
!review

File: code/game/objects/structures/undetonated_nuke.dm [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	switch(severity)
+		if(1.0)
+			explode()
+			return
+		if(2.0)
+			if(prob(50))
+				explode()
+				return
+		if(3.0)
+			return
```
Comment (DreamySkrell) [inline code/game/objects/structures/undetonated_nuke.dm:27]: what if severity is 1.5 or 2.5 or any other number that is not exactly 1.0 or 2.0 or 3.0

**Review by DreamySkrell — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/undetonated_nuke.dm (hunk 1 above)
Comment (DreamySkrell) [reply code/game/objects/structures/undetonated_nuke.dm:27]: it won't explode then

File: code/game/objects/structures/crates_lockers/closets/gimmick.dm [hunk 2]
```diff
@@ -134,3 +134,82 @@
...
+	icon_state = "wc_sarcophagus"
+	dense_when_open = TRUE
+	anchored = TRUE
+	canbemoved = FALSE
+	open_sound = 'sound/effects/stonedoor_openclose.ogg'
+	close_sound = 'sound/effects/stonedoor_openclose.ogg'
+	var/open_state = "wc_sarcophagus_open"
+	var/closed_state = "wc_sarcophagus"
+	var/trapped = FALSE
+	var/triggered = FALSE
```
Comment (FluffyGhoster) [inline code/game/objects/structures/crates_lockers/closets/gimmick.dm:151]: DMDoc

File: code/game/objects/structures/crates_lockers/closets/gimmick.dm [hunk 3]
```diff
@@ -134,3 +134,82 @@
...
+		icon_state = closed_state
+
+	else
+		layer = BELOW_OBJ_LAYER
+		icon_state = open_state
+
+/obj/structure/closet/sarcophagus/animate_door(var/closing = FALSE)
+	return
+
+/obj/structure/closet/sarcophagus/attack_hand(mob/user as mob)
```
Comment (FluffyGhoster) [inline code/game/objects/structures/crates_lockers/closets/gimmick.dm:165]:
```suggestion
/obj/structure/closet/sarcophagus/attack_hand(mob/user)
```

File: code/game/objects/structures/crates_lockers/closets/gimmick.dm [hunk 4]
```diff
@@ -134,3 +134,82 @@
...
+	..()
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(trapped && !triggered)
+			do_trap_effect(H)
+
+/obj/structure/closet/sarcophagus/proc/do_trap_effect(var/mob/living/carbon/human/H)
```
Comment (FluffyGhoster) [inline code/game/objects/structures/crates_lockers/closets/gimmick.dm:172]: DMDoc

File: code/game/objects/structures/crates_lockers/closets/gimmick.dm [hunk 5]
```diff
@@ -134,3 +134,82 @@
...
+	if(triggered)
+		return
+	var/trap = pick("arrow", "fire", "poison")
+	switch(trap)
```
Comment (FluffyGhoster) [inline code/game/objects/structures/crates_lockers/closets/gimmick.dm:176]: Pointless use of strings, use probability

File: code/game/objects/structures/undetonated_nuke.dm [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+/obj/structure/undetonated_nuke/proc/explode() //Sets off a big conventional explosion from the trigger and scatters radioactive material, but doesn't do a full-on nuclear explosion because nukes are not video game explosive barrels.
```
Comment (FluffyGhoster) [inline code/game/objects/structures/undetonated_nuke.dm:29]: DMDoc

File: code/game/objects/structures/undetonated_nuke.dm [hunk 7]
```diff
@@ -0,0 +1,59 @@
+/obj/structure/undetonated_nuke
+	name = "undetonated nuclear bomb"
+	desc = "This rusted bomb looks like it's been here for decades. Flaking paint on the side depicts the battle standard of the Traditionalist Coalition. Something is written in Sinta'Azaziba on the side."
+	icon = 'icons/obj/unathi_ruins.dmi'
+	icon_state = "bomb"
+	density = TRUE
+	anchored = FALSE
+	var/can_explode = TRUE
```
Comment (FluffyGhoster) [inline code/game/objects/structures/undetonated_nuke.dm:8]: DMDoc

File: code/game/objects/structures/undetonated_nuke.dm [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+/obj/structure/undetonated_nuke/buried/Initialize(mapload)
+	. = ..()
+	if(prob(50))
+		can_explode = FALSE
+
+/obj/structure/undetonated_nuke/buried/attackby(obj/item/attacking_item, mob/user, params)
+	. = ..()
+	if(istype(get_turf(src), /turf/simulated/floor/exoplanet))
+		var/turf/simulated/floor/exoplanet/T = get_turf(src)
```
Comment (FluffyGhoster) [inline code/game/objects/structures/undetonated_nuke.dm:51]: Pointless doublecall to get_turf

File: code/game/objects/structures/undetonated_nuke.dm [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+		if(attacking_item.is_shovel() && T.diggable)
+			visible_message(SPAN_NOTICE("\The [user] begins to excavate \the [src]."))
+			if(attacking_item.use_tool(src, user, 100, volume = 50))
+				visible_message(SPAN_NOTICE("\The [user] finishes digging \the [src] from \the [T]!"))
+				new /obj/structure/pit(T)
+				var/obj/structure/undetonated_nuke/N = (T)
```
Comment (FluffyGhoster) [inline code/game/objects/structures/undetonated_nuke.dm:57]:
```suggestion
				var/obj/structure/undetonated_nuke/N = T
```

File: code/game/objects/structures/undetonated_nuke.dm (hunk 1 above)
Comment (RustingWithYou) [reply code/game/objects/structures/undetonated_nuke.dm:27]: Severity for ex_act is rounded to a whole number from 1-3 IIRC, and this is pretty much identical to how most other ex_acts in the game are handled

File: code/game/objects/structures/crates_lockers/closets/gimmick.dm [hunk 10]
```diff
@@ -134,3 +134,85 @@
...
+		return
+	if(prob(33))
+		var/turf/T = get_turf(src)
+		var/obj/item/arrow/arrow = new(T)
+		playsound(usr.loc, 'sound/weapons/crossbow.ogg', 75, 1)
+		arrow.throw_at(H, 10, 9, src) //same values as a full draw crossbow shot would have
+
+	else if(prob(33))
+		visible_message(SPAN_DANGER("Flames engulf \the [H]!"))
+		H.adjustFireLoss(30)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/crates_lockers/closets/gimmick.dm:188]: use a proper apply_damage call here

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/undetonated_nuke.dm (hunk 1 above)
Comment (DreamySkrell) [reply code/game/objects/structures/undetonated_nuke.dm:28]: fair

**Review by DreamySkrell — APPROVED** [review]

File: code/game/objects/structures/crates_lockers/closets/gimmick.dm (hunk 10 above)
Comment (RustingWithYou) [reply code/game/objects/structures/crates_lockers/closets/gimmick.dm]: done

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**